spock-specs/src/test/groovy/org/spockframework/smoke/parameterization/DataTables.groovy

```diff
@@ -166,6 +247,29 @@ a | a
     d = c + 1
   }
 
+  def "tables with semicolon can be mixed with other parameterizations"() {
+    expect:
+    [a, b, c, d] == [1, 2, 3, 4]
+
+    where:
+    a << [1]
+    b ; c
+    2 ; 3
+    d = c + 1
+  }
+
+  def "tables with different separators can be mixed"() {
```

Comment:
I'm missing a test that describes your original problem with using `|` in a data table value